@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -398,48 +437,84 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
                 text = String(text);
 
                 if (this.m_skipShortLabels) {
-                    // Filter the lines, keep only those that are long enough for labelling.
+                    // Filter the lines, keep only those that are long enough for labelling. Also,
+                    // split jagged label paths to keep processing and rendering only those that
+                    // have no sharp corners, which would not be rendered anyway.
                     validLines = [];
 
                     const metersPerPixel = tileSizeInMeters / this.m_decodeInfo.tileSizeOnScreen;
-                    const minTileSpace =
+                    const minEstimatedLabelLength =
                         MIN_AVERAGE_CHAR_WIDTH *
                         text.length *
                         metersPerPixel *
                         SIZE_ESTIMATION_FACTOR;
+                    const minEstimatedLabelLengthSqr =
```

Comment:
I understand the need to not want to have functions, but this code makes a function that is already hard to read much worse.

Clean code says we should have a single level of abstraction in each function (http://www.principles-wiki.net/principles:single_level_of_abstraction) , whilst I don't think we need to go that far, this function now has 7 layers of indentation, which you can say is roughly 7 levels of abstraction (in a very coarse sense).

Please add some methods to remove the large jumps in abstraction (or perhaps convince me otherwise).

Bonus points however for computeBoundingBoxSizeSqr.

Reply:
Extracted code into protected method.